### DISCUSSION TITLE
Tuya switch stops responding to ARP requests

### DISCUSSION THREAD

**Discussion by DSTB-UK** [jasonacox/tinytuya#317] [General]
Hi, I've having a stange issue with a Tuya switch. I can communicate with the switch fine using TinyTuya, but after a few hours / days I start getting Network Error: Device Unreachable responses from TinyTuya. The switch still works ok from the smart life app via the cloud. A reboot of the switch fixes the issue but it's not ideal. 
A quick wireshark showed the switch was no longer repsonding to ARP discovery requests. Adding a static arp entry in Windows / Linux fixes the issue. I'm guessing the cloud control still worked as the network router(network switch) sees the regular UDP packets and adds it to it's arp cache without ever needing to send an ARP dicovery.
Just wondered if anyone else had experiance this or found a more elegant solution the adding static arp entries.
  Cheers

**Comment by jasonacox** [comment]
Hi @DSTB-UK  - I have two older 3.1 devices ([TanTan Smart Plug](https://www.ebay.com/itm/266083605443?hash=item3df3d19bc3:g:888AAOSwjv1jwat5&amdata=enc%3AAQAHAAAA4FGAdtw3D3%2FO0xwz3cusH5TzofPjIfMuNSVh75cJLv9Hkb62B0eRZG%2BIlPcD%2B%2FoHRLeJkKZ1YHCcBqgNLUhMJZ3K5qxAg5XEWJX7ZFAIwb5e29P9K1M1ldvM%2BiHACpN1cSQVacxR%2Bx91vJiQbiHGrQysjq11dyMBaPrU45vVtYK%2BhuaaFbyZitY700joguPsKGN3ojsyXetsIL9Z1RTovQ0IeZ3AV2t0QGMbUWjKZiQCkOTpHHLXtPVzirG08jZb5JytouoKcMVnhShbMi78%2FnalhoVV2uJIt3Wl%2Bw1djvIA%7Ctkp%3ABk9SR6awtLPnYQ)) that do this. I suspect they have a buggy firmware (no updates available) that causes it to stop processing local requests but keeps connected to the Cloud. A power cycle fixes mine as well. I don't use them anymore for anything but TinyTuya testing.  Can you post a link and version to the devices you have?

**Reply by DSTB-UK** [reply]
The device is an [XUELILI Smart Switch](https://www.amazon.co.uk/dp/B09P88YF2W?psc=1&ref=ppx_yo2ov_dt_b_product_details) 
It's running version 3.3. Would be interesting to see if your switches have the same issue i.e. when you lose local connection try statically adding the mac to the ARP table. Could probably be done programmatically but seems a lot of work for an uncommon issue. Didn't find any reference to this issue when googling it.
And thanks for the awesome TinyTuya, great app and really easy to follow guides. Appreciate all your work

**Reply by jasonacox** [reply]
Thanks, @DSTB-UK !  

> statically adding the mac to the ARP table.

I'll give that a try.
